src/core/parser/TommyParser.kt

```diff
@@ -240,10 +238,16 @@ class TommyParser : Grammar<List<AST>>() {
     //else
         //return "no"
     //end
-    val ifParser = -IF and expr and -THEN and zeroOrMore(parser(this::astParser)) and
-             zeroOrMore(-ELSEIF and expr and -THEN and zeroOrMore(parser(this::astParser))) and optional(-ELSE and zeroOrMore(parser(this::astParser))) and -END map { (cond, body, elifs, elsebody) ->
-        If(cond, body, elsebody, elifs)
-    }
+    private val elseifParser = -ELSEIF and expr and -THEN and zeroOrMore(parser(this::astParser))
+    private val ifParser = -IF and expr and -THEN and zeroOrMore(parser(this::astParser)) and
+             zeroOrMore(elseifParser) and optional(-ELSE and zeroOrMore(parser(this::astParser))) and
+            -END map { (cond, body, elifs, elsebody) ->
+                val elseBranch : If? = elsebody?.let { Else(it) }
+                val elseIfs = elifs.foldRight(elseBranch) { (currCond, currBody), next ->
```

Comment:
clever, if parsing is much cleaner now

Reply:
yeah I didn't like  how we used List<Tuple2<Expr, Body>> for the elseifs and I figured we could do a pseudo-linked list for ifs instead